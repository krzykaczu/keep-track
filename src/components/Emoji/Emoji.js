import React from 'react';
import PropTypes from 'prop-types';

const Emoji = ({ label, symbol }) => (
  <span
    role="img"
    aria-label={label ? label : ''}
    aria-hidden={label ? 'false' : 'true'}
    data-testid="emoji"
  >
    {symbol}
  </span>
);

Emoji.propTypes = {
  label: PropTypes.string,
  symbol: PropTypes.string.isRequired,
};

export default Emoji;
