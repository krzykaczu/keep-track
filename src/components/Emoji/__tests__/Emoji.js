import React from 'react';
import Emoji from '..';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup } from 'react-testing-library';

afterEach(cleanup);

test('Emoji rendered properly', () => {
  const { getByTestId, container } = render(<Emoji symbol="➕" label="add" />);
  expect(container).toMatchSnapshot();
  expect(getByTestId('emoji')).toHaveAttribute('role', 'img');
  expect(getByTestId('emoji')).toHaveAttribute('aria-label', 'add');
  expect(getByTestId('emoji')).toHaveAttribute('aria-hidden', 'false');
  expect(getByTestId('emoji')).toHaveTextContent('➕');
});

test('Emoji with no label rendered properly', () => {
  const { getByTestId } = render(<Emoji symbol="➕" />);
  expect(getByTestId('emoji')).toHaveAttribute('aria-label', '');
  expect(getByTestId('emoji')).toHaveAttribute('aria-hidden', 'true');
  expect(getByTestId('emoji')).toHaveTextContent('➕');
});
