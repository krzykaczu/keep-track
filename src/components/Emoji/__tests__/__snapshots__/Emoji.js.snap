// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji rendered properly 1`] = `
<div>
  <span
    aria-hidden="false"
    aria-label="add"
    data-testid="emoji"
    role="img"
  >
    âž•
  </span>
</div>
`;
